File: ADSystem/Scripts/Class/Commissions/Commissions.js
angular.module("General").controller('CommissionsController', function (GUID, $scope, $http, $window, notify, $rootScope) {

    $scope.Commissions = new Array();
    $scope.idUser = 0;

    $scope.Init = function (idUser) {
        $scope.idUser = idUser;
        $scope.GetCommissions();
    };

    $scope.GetCommissions = function () {
        $http({
            method: 'GET',
            url: '../../../Commissions/Get',
            params: {
                idUser: $scope.idUser
            }
        }).
        success(function (data, status, headers, config) {
            if (data.success == 1) {
                $scope.Commissions = [];
                if (data.oData.Commissions.length > 0) {
                    angular.forEach(data.oData.Commissions, function (value, key) {
                        $scope.Commissions.push({
                            idComision: value.idComision,
                            idUsuario: value.idUsuario,
                            LimiteInferior: value.LimiteInferior,
                            LimiteSuperior: value.LimiteSuperior,
                            SueldoMensual: value.SueldoMensual,
                            PorcentajeComision: value.PorcentajeComision,
                            SueldoComision: value.SueldoComision,
                            BonoUno: value.BonoUno,
                            BonoDos: value.BonoDos,
                            BonoTres: value.BonoTres
                        });
                    });
                } else {
                    notify('No se encontraron registros.', $rootScope.error);
                }
            } else if (data.failure == 1) {
                notify(data.oData.Error, $rootScope.error);
            } else if (data.noLogin == 1) {
                window.location = "../../../Access/Close";
            }
        }).
        error(function (data, status, headers, config) {
            notify("Ocurrío un error.", $rootScope.error);
        });
    };

    $scope.Add = function () {
        $http({
            method: 'POST',
            url: '../../../Commissions/Add',
            params: {
                idUsuario: $scope.idUser,
                LimiteInferior: $scope.LimiteInferior,
                LimiteSuperior: $scope.LimiteSuperior,
                SueldoMensual: $scope.SueldoMensual,
                PorcentajeComision: $scope.PorcentajeComision,
                SueldoComision: 0,
                BonoUno: $scope.BonoUno,
                BonoDos: $scope.BonoDos,
                BonoTres: $scope.BonoTres
            }
        }).
        success(function (data, status, headers, config) {
            if (data.success == 1) {
                notify(data.oData.Message, $rootScope.success);
                $scope.CleanCommission();
                $scope.GetCommissions();
            } else if (data.failure == 1) {
                notify(data.oData.Error, $rootScope.error);
            } else if (data.noLogin == 1) {
                window.location = "../../../Access/Close";
            }
        }).
        error(function (data, status, headers, config) {
            notify("Ocurrío un error.", $rootScope.error);
        });
    };

    $scope.Update = function (commission) {
        $http({
            method: 'PATCH',
            url: '../../../Commissions/Update',
            params: {
                idComision: commission.idComision,
                idUsuario: $scope.idUser,
                LimiteInferior: commission.LimiteInferior,
                LimiteSuperior: commission.LimiteSuperior,
                SueldoMensual: commission.SueldoMensual,
                PorcentajeComision: commission.PorcentajeComision,
                SueldoComision: 0,
                BonoUno: commission.BonoUno,
                BonoDos: commission.BonoDos,
                BonoTres: commission.BonoTres
            }
        }).
        success(function (data, status, headers, config) {
            if (data.success == 1) {
                $scope.CleanCommission();
                $scope.GetCommissions();
            } else if (data.failure == 1) {
                notify(data.oData.Error, $rootScope.error);
            } else if (data.noLogin == 1) {
                window.location = "../../../Access/Close";
            }
        }).
        error(function (data, status, headers, config) {
            notify("Ocurrío un error.", $rootScope.error);
        });
    };

    $scope.Remove = function (idCommission) {
        $http({
            method: 'DELETE',
            url: '../../../Commissions/Delete',
            params: {
                idCommission: idCommission
            }
        }).
            success(function (data, status, headers, config) {
                if (data.success == 1) {
                    $scope.CleanCommission();
                    $scope.GetCommissions();
                } else if (data.failure == 1) {
                    notify(data.oData.Error, $rootScope.error);
                } else if (data.noLogin == 1) {
                    window.location = "../../../Access/Close";
                }
            }).
            error(function (data, status, headers, config) {
                notify("Ocurrío un error.", $rootScope.error);
            });
    };

    $scope.CleanCommission = function () {
        $scope.LimiteInferior = 0;
        $scope.LimiteSuperior = 0;
        $scope.SueldoMensual = 0;
        $scope.PorcentajeComision = 0;
        $scope.BonoUno = 0;
        $scope.BonoDos = 0;
        $scope.BonoTres = 0;
    }
});

File: ADEntities/Queries/Commissions.cs
using ADEntities.Models;
using ADEntities.ViewModels;
using System;
using System.Collections.Generic;
using System.Data.Entity.Validation;
using System.Linq;
using System.Web;

namespace ADEntities.Queries
{
    public class Commissions: Base
    {
        public CommissionViewModel GetById(int idCommission)
        {
            using (var context = new admDB_SAADDBEntities())
                try
                {
                    return context.tComisiones.Where(p=>p.idComision == idCommission).Select(p => new CommissionViewModel()
                    {
                        idComision = p.idComision,
                        idUsuario = p.idUsuario,
                        LimiteInferior = p.LimiteInferior,
                        LimiteSuperior = p.LimiteSuperior,
                        SueldoMensual = p.SueldoMensual,
                        PorcentajeComision = p.PorcentajeComision,
                        SueldoComision = p.SueldoComision,
                        BonoUno = p.BonoUno,
                        BonoDos = p.BonoDos,
                        BonoTres = p.BonoTres
                    }).FirstOrDefault();
                }
                catch (DbEntityValidationException ex)
                {
                    var newException = new ADEntities.Common.FormattedDbEntityValidationException(ex);
                    throw newException;
                }
        }

        public List<CommissionViewModel> GetCommissions(int idUser)
        {
            using (var context = new admDB_SAADDBEntities())
                try
                {
                    return context.tComisiones.Where(p=>p.idUsuario == idUser).Select(p => new CommissionViewModel()
                    {

                        idComision = p.idComision,
                        idUsuario = p.idUsuario,
                        LimiteInferior = p.LimiteInferior,
                        LimiteSuperior = p.LimiteSuperior,
                        SueldoMensual = p.SueldoMensual,
                        PorcentajeComision = p.PorcentajeComision,
                        SueldoComision = p.SueldoComision,
                        BonoUno = p.BonoUno,
                        BonoDos = p.BonoDos,
                        BonoTres = p.BonoTres

                    }).OrderBy(p => p.idComision).ToList();
                }
                catch (DbEntityValidationException ex)
                {
                    var newException = new ADEntities.Common.FormattedDbEntityValidationException(ex);
                    throw newException;
                }
        }

        public int Add(tComisione commission)
        {
            using (var context = new admDB_SAADDBEntities())
                try
                {
                    context.tComisiones.Add(commission);

                    return context.SaveChanges();
                }
                catch (DbEntityValidationException ex)
                {
                    var newException = new ADEntities.Common.FormattedDbEntityValidationException(ex);
                    throw newException;
                }
        }        

        public int Update(tComisione commission)
        {
            using (var context = new admDB_SAADDBEntities())
                try
                {
                    var entity = context.tComisiones.Where(p => p.idComision == commission.idComision).FirstOrDefault();

                    entity.LimiteInferior = commission.LimiteInferior;
                    entity.LimiteSuperior = commission.LimiteSuperior;
                    entity.SueldoMensual = commission.SueldoMensual;
                    entity.PorcentajeComision = commission.PorcentajeComision;
                    entity.SueldoComision = commission.SueldoComision;
                    entity.BonoUno = commission.BonoUno;
                    entity.BonoDos = commission.BonoDos;
                    entity.BonoTres = commission.BonoTres;

                    return context.SaveChanges();
                }
                catch (DbEntityValidationException ex)
                {
                    var newException = new ADEntities.Common.FormattedDbEntityValidationException(ex);
                    throw newException;
                }
        }

        public void Remove(int idCommission)
        {
            using (var context = new admDB_SAADDBEntities())
                try
                {
                    var entity = context.tComisiones.Where(p => p.idComision == idCommission).FirstOrDefault();

                    context.tComisiones.Remove(entity);

                    context.SaveChanges();
                }
                catch (DbEntityValidationException ex)
                {
                    var newException = new ADEntities.Common.FormattedDbEntityValidationException(ex);
                    throw newException;
                }
        }
    }
}